app/models/supplejack_api/support/searchable.rb

```diff
@@ -76,21 +76,34 @@ def build_sunspot_schema(builder)
 
             options[:as] = field.solr_name if field.solr_name.present?
 
-            if search_as.include? :filter
-              filter_options = {}
-              filter_options[:multiple] = true if field.multi_value.present?
-              type = SUNSPOT_TYPE_NAMES[field.type]
+            add_filter_to(builder, field, options, value_block) if search_as.include? :filter
 
-              builder.public_send(type, field.name, options.merge(filter_options), &value_block)
-            end
+            add_fulltext_to(builder, field, options, value_block) if search_as.include? :fulltext
 
-            if search_as.include? :fulltext
-              options[:boost] = field.search_boost if field.search_boost.present?
-              builder.text field.name, options, &value_block
-            end
+            add_mlt_to(builder, field, options, value_block) if search_as.include? :mlt
```

Comment:
This is the new addition

Reply:
Is this a very heavy process? We currently have issues with performance with the fulltext field.

Reply:
It will be similar. For PVM indexing with this was not bad at all.

Reply:
Nice thanks for having a look :)